src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int START_CAPACITY = 16;
+    private static final int ARRAY_EXPANSION = 2;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private Node<K, V>[] heads;
+    private int capacity;
+    private int size;
+
+    public MyHashMap() {
+        capacity = START_CAPACITY;
+        heads = new Node[START_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        increaseCapacity();
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value);
+        if (heads[index] == null) {
+            heads[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> tmp = heads[index];
+        if (Objects.equals(tmp.key, key)) {
+            tmp.value = value;
+            return;
+        }
+        Node prev = null;
+        while (tmp != null) {
+            if (Objects.equals(tmp.key, key)) {
+                tmp.value = newNode.value;
+                return;
+            }
+            prev = tmp;
+            tmp = tmp.next;
+        }
+        prev.next = newNode;
+        tmp = newNode;
+        size++;
 
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> tmp = heads[getIndex(key)];
+        while (tmp != null) {
+            if (Objects.equals(key, tmp.key)) {
+                return tmp.value;
+            }
+            tmp = tmp.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void increaseCapacity() {
```

Comment:
let's rename to resize(), it's more popular name for this function
![image](https://user-images.githubusercontent.com/65075884/184877360-33937fdc-c7f2-44b5-b80c-55fa51e6dfcc.png)